src/rt/reactor.rs

```diff
@@ -104,17 +120,19 @@ impl Reactor {
         vacant.insert(entry.clone());
 
         // Register the I/O event source in the poller.
-        let interest = mio::Ready::all();
-        let opts = mio::PollOpt::edge();
-        self.poller.register(source, token, interest, opts)?;
+        self.poller
+            .lock()
+            .unwrap()
+            .registry()
+            .register(source, token, INTEREST_ALL)?;
 
         Ok(entry)
     }
 
     /// Deregisters an I/O event source associated with an entry.
-    fn deregister(&self, source: &dyn Evented, entry: &Entry) -> io::Result<()> {
+    fn deregister(&self, source: &mut dyn Source, entry: &Entry) -> io::Result<()> {
         // Deregister the I/O object from the mio instance.
-        self.poller.deregister(source)?;
+        self.poller.lock().unwrap().registry().deregister(source)?;
```

Comment:
this seems wrong - register/deregister will have to wait until any current poll finishes, to acquire the lock?

Reply:
[More information on this](https://github.com/tokio-rs/mio/pull/932). Discussion is [here](https://github.com/tokio-rs/mio/issues/757).